src/ApparentHorizons/Strahlkorper.hpp

```diff
@@ -0,0 +1,129 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+#include "ApparentHorizons/YlmSpherepack.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ForceInline.hpp"
+
+/// \ingroup SurfacesGroup
+/// \brief A star-shaped surface expanded in spherical harmonics.
+template <typename Frame>
+class Strahlkorper {
+ public:
+  // Pup needs default constructor
+  Strahlkorper() = default;
+
+  /// Construct a sphere of radius r with a given center.
+  Strahlkorper(size_t l_max, size_t m_max, double radius,
+               std::array<double, 3> center) noexcept;
+  /// Prolong or restrict another surface to the given l_max and m_max.
+  Strahlkorper(size_t l_max, size_t m_max,
+               const Strahlkorper& another_strahlkorper) noexcept;
+  /// Construct a Strahlkorper from another Strahlkorper,
+  /// but explicitly specifying the coefficients.
+  /// Here coefficients are in the same storage scheme
+  /// as the `coefficients` member function returns.
+  Strahlkorper(const Strahlkorper& another_strahlkorper,
+               DataVector coefs) noexcept;
+
+  /// Move-construct a Strahlkorper from another Strahlkorper,
+  /// explicitly specifying the coefficients.
+  Strahlkorper(Strahlkorper&& another_strahlkorper, DataVector coefs) noexcept;
+
+  /// Construct a Strahlkorper from a DataVector containing the radius
+  /// at the collocation points.
+  Strahlkorper(const DataVector& radius_at_collocation_points, size_t l_max,
+               size_t m_max, std::array<double, 3> center) noexcept;
+
+  // clang-tidy: no runtime references
+  /// Serialization for Charm++
+  void pup(PUP::er& p);  // NOLINT
+
+  /*!
+   *  These coefficients are stored as SPHEREPACK coefficients.
+   *  Suppose you represent a set of coefficients \f$F^{lm}\f$ in the expansion
+   *  \f[
+   *  f(\theta,\phi) =
+   *  \sum_{l=0}^{l_max} \sum_{m=-l}^{l} F^{lm} Y^{lm}(\theta,\phi)
+   *  \f]
+   *  Here the \f$Y^{lm}(\theta,\phi)\f$ are the usual complex-valued scalar
+   *  spherical harmonics, so \f$F^{lm}\f$ are also complex-valued.
+   *  But here we assume that \f$f(\theta,\phi)\f$ is real, so therefore
+   *  the \f$F^{lm}\f$ obey \f$F^{l-m} = (-1)^m (F^{lm})^\star\f$, so one
+   *  does not need to store both real and imaginary parts.
+   *  So:
+   *  strahlkorper_coefs_(l,m) = \f$(-1)^m Q Re(F^{lm})\f$ for \f$m\ge 0\f$
+   *  strahlkorper_coefs_(l,m) = \f$(-1)^m Q Im(F^{lm})\f$ for \f$m<   0\f$
```

Comment:
Perhaps we should avoid referring to `strahlkorper_coefs_(l,m)`, a member variable, in the Doxygen comment? It may not be obvious what this is. One alternative, which refers to the function name directly: `coefficients()(l,m)`